src/decoders/sndfile.cpp

```diff
@@ -226,6 +237,15 @@ SharedPtr<Decoder> SndFileDecoderFactory::createDecoder(UniquePtr<std::istream>
     SampleType stype = SampleType::Int16;
     switch(sndinfo.format&SF_FORMAT_SUBMASK)
     {
+        case SF_FORMAT_PCM_U8:
+          stype = SampleType::UInt8;
+          break;
+        case SF_FORMAT_ULAW:
+            if(Context::GetCurrent().isSupported(sconfig, SampleType::Mulaw))
+            {
+                stype = SampleType::Mulaw;
+                break;
+            }
```

Comment:
This will fall through to trying `Float32` if `Mulaw` isn't available, which then falls through to `Int16`. It would be better to fall back to `Int16` directly if `Mulaw` isn't supported instead.